includes/Classifai/Providers/Azure/ComputerVision.php

```diff
@@ -801,7 +801,9 @@ protected function generate_alt_tags( $captions, $attachment_id ) {
 					update_post_meta( $attachment_id, '_wp_attachment_image_alt', $captions[0]->text );
 				}
 
-				if ( in_array( 'caption', $enabled_fields, true ) ) {
+				$excerpt = get_the_excerpt( $attachment_id );
+
+				if ( in_array( 'caption', $enabled_fields, true ) && $excerpt !== $captions[0]->text ) {
```

Comment:
I don't think this truly fixes the issue, though I didn't test. But in the scenario where an image doesn't have an excerpt set and you trigger the generation of one, this check will return `false` (which is correct) and `wp_update_post` will be called, which then triggers this function to be called again. The second time it runs this code check will return `true` and so things won't execute again but ideally we should never execute this code again in the first place.

I'd suggest instead of doing this we remove the hook if we make it to the `wp_update_post` part, so it doesn't trigger again in that execution:

`remove_action( 'edit_attachment', [ $this, 'maybe_rescan_image' ] );`

Reply:
In addition to the above, I think there are a few other places where we have the same issue (again, just looking at the code, didn't test this).

I see a `wp_update_post` call just a few lines below this (line 814) and then our OCR processing I think will have the same issue as we call `wp_update_post` in our `generate_ocr_data` method, which can be triggered by the `edit_attachment` hook